server/incore-common/src/main/java/edu/illinois/ncsa/incore/common/utils/AllocationUtils.java

```diff
@@ -0,0 +1,260 @@
+/*******************************************************************************
+ * Copyright (c) 2022 University of Illinois and others.  All rights reserved.
+ * This program and the accompanying materials are made available under the
+ * terms of the Mozilla Public License v2.0 which accompanies this distribution,
+ * and is available at https://www.mozilla.org/en-US/MPL/2.0/
+ *
+ *   Contributors:
+ *   Yong Wook Kim (NCSA) - initial API and implementation
+ *******************************************************************************/
+
+package edu.illinois.ncsa.incore.common.utils;
+
+import edu.illinois.ncsa.incore.common.AllocationConstants;
+import edu.illinois.ncsa.incore.common.dao.IUserFinalQuotaRepository;
+import edu.illinois.ncsa.incore.common.exceptions.IncoreHTTPException;
+import edu.illinois.ncsa.incore.common.models.UserAllocations;
+import edu.illinois.ncsa.incore.common.models.UserFinalQuota;
+import edu.illinois.ncsa.incore.common.models.UserUsages;
+import edu.illinois.ncsa.incore.common.dao.IUserAllocationsRepository;
+
+
+import javax.ws.rs.core.Response;
+
+public class AllocationUtils {
+
+    /***
+     * This method receives a username and to see if the user's number of dataset is within the allocation.
+     *
+     * @param allocationRepository IUserAllocationsRepository injected by controller
+     * @param quotaRepository IUserFinalQuotaRepository injected by controller
+     * @param username string representation of username
+     * @param datasetType string for representing the type of the dataset
+     * @return postOk boolean if the user can post more datasets
+     */
+    public static Boolean canCreateAnyDataset(IUserAllocationsRepository allocationRepository,
+                                           IUserFinalQuotaRepository quotaRepository,
+                                           String username, String datasetType) {
```

Comment:
can we rename the `datasetType` argument to be `entityType`, so it(dfr3/hazard etc.) is not confused with incore datasets.

Reply:
modified